src/pages/Products/Product.js

```diff
@@ -59,6 +64,18 @@ export const Product = () => {
 		})();
 	}, []);
 
+	const { cartState, cartDispatch } = useCart();
+	const { token } = useAuth();
+	const navigate = useNavigate();
+
+	const addToCartHandler = async (product) => {
+		if (token) {
+			addToCart(product, cartDispatch, token);
```

Comment:
`addToCart` is already an async function you can remove the async keyword from `addToHandler` function.